tftbase.py

```diff
@@ -21,6 +22,7 @@
 
 logger = logging.getLogger("tft." + __name__)
 
+tft_lock = threading.Lock()
```

Comment:
Why do we need this lock? What is being thread here during config?

Reply:
hm...

well, with this PR, registration happens when the modules are imported (and call `pluginbase.register_plugin()` and `TestTypeHandler.register_test_type()`). There is also only one place when we import the modules, and that is during `pluginbase._get_registry()` and `TestTypeHandler.get()` (although, the code really should also work if you import the modules manually and explicitly).

ocp-tft is multi threaded. Still, I would very much suspect that above getters are called before other threads start, so most likely there should not be a problem. At least "today". This is the important part, because it adds hidden requirements like "ensure that the modules are imported on a thread before other threads are started". We have a mult-threaded application, and this plugin/module registration necessarily uses global variables. I think there should be a lock around accessing them just as a matter of principle.

---

Usually, when we want to lock one thing (e.g. access to `pluginbase._registry`), then we create a dedicated lock instance. In this case, I instead reused the same lock `tftbase.tft_lock` for different purposes. I think it's fine because:
- whenever we hold a lock we always must ensure that we don't call into unknown code (e.g. use provided callbacks), so that we don't take the same lock again (`threading.lock` is not reentrant) and don't have multiple locks that are taken in undefined order (danger of deadlock). In our use here, it's easy to see that we don't call to code that could lead to reentrantly taking a lock or to deadlock.
- using the same lock might decrease parallelism, but in practice, this doesn't matter here.

---

Anyway. I drop `tftbase.tft_lock` and add instead two specific locks per usage. That makes it clearer for when each lock is used.